systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,191 @@
+        LOGGER.info("====================================================================================");
+        LOGGER.info("============== Verification version of CO:" + fromVersion + " => " + toVersion);
+        LOGGER.info("====================================================================================");
+
+        kafkaBasicClientJob.producerStrimzi().done();
+        kafkaBasicClientJob.consumerStrimzi().done();
+
+        // ======== Cluster Operator upgrade starts ========
+        Map<String, String> kafkaSnapshot = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        // wait until non-used install plan is present (sometimes install-plan did not append immediately and we need to wait for at least 10m)
+        OlmUtils.waitUntilNonUsedInstallPlanIsPresent(toVersion);
+
+        // Cluster Operator
+        OlmResource.upgradeClusterOperator();
+
+        // wait until RU is finished (first run skipping)
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaSnapshot);
+        // ======== Cluster Operator upgrade ends ========
+
+        // verification that cluster operator has correct version (install-plan) - strimzi-cluster-operator.v[version]
+        String afterUpgradeVersionOfCo = OlmResource.getClusterOperatorVersion();
+
+        // if HEAD -> 6.6.6 version
+        toVersion = toVersion.equals("HEAD") ? Environment.OLM_LATEST_CONTAINER_IMAGE_TAG_DEFAULT : toVersion;
+        assertThat(afterUpgradeVersionOfCo, is(Environment.OLM_APP_BUNDLE_PREFIX + ".v" + toVersion));
+
+        // ======== Kafka upgrade starts ========
+        // Make snapshots of all pods
+        makeSnapshots();
+        logPodImages();
+        changeKafkaAndLogFormatVersion(testParameters.getJsonObject("proceduresAfter"));
+        logPodImages();
+        // ======== Kafka upgrade ends ========
+
+        ClientUtils.waitForClientSuccess(producerName, namespace, messageUpgradeCount);
+        ClientUtils.waitForClientSuccess(consumerName, namespace, messageUpgradeCount);
+
+        // Delete jobs to make same names available for next upgrade
+        kubeClient().deleteJob(producerName);
+        kubeClient().deleteJob(consumerName);
+
+        // Check errors in CO log
+        assertNoCoErrorsLogged(0);
+
+        // Save install-plan to closed-map
+        OlmResource.getClosedMapInstallPlan().put(OlmResource.getNonUsedInstallPlan(), Boolean.TRUE);
+    }
+
+    /**
+     * Loads auxiliary information from StrimziUpgradeST.json
+     * [0] -> from version
+     * [1] -> to version
+     * [2] -> whole JsonObject for first supported version
+     * @param indexOfItem specific index which you want to access
+     * @return exception || first supported version || json object with upgrade information
+     */
+    private Object getFirstSupportedItemFromUpgradeJson(int indexOfItem) {
```

Comment:
Wonder if adding some metadata to `environment` in ugprade json wouldn't be better solution